testng-core/src/main/java/org/testng/TestNG.java

```diff
@@ -1130,14 +1138,21 @@ protected List<ISuite> runSuites() {
   }
 
   private void runSuiteAlterationListeners() {
-    for (IAlterSuiteListener l : m_alterSuiteListeners.values()) {
+    List<IAlterSuiteListener> original = Lists.newArrayList(m_alterSuiteListeners.values());
+    Optional.ofNullable(m_configuration.getListenerComparator())
```

Comment:
Consider having a default `ListenerComparator` and you'll be able to remove all null checks.

Reply:
I dont know if a comparator such as `(a,b) -> 0` would be a valid comparator that ensures that the original insertion order is preserved.

Reply:
The other concern I have is that I would like to sort `if and only if` there was a comparator that was provided by the user. Else there's no point in trying to sort something in the original order. Wouldn't that be redundant ?

Reply:
In fact, having a default comparator is not a good idea because it will sort every time for nothing in case of default.